### ISSUE TITLE
[gatsby-plugin-mdx]: browserslist config not working when using gatsby-plugin-mdx

### ISSUE THREAD

**Issue by Enter-tainer** [gatsbyjs/gatsby#25003]
## Description

**TL;DR: When gatsby-plugin-mdx is enabled, it will not respect `browserslist` config in `package.json`.**

I'm customizing browserslist. And I followed the instruction in [Browser Supper | GatsbyJS](https://www.gatsbyjs.org/docs/browser-support/#specify-what-browsers-your-project-supports-using-browserslist).

For test purpose, I have set the `browserslist` in `package.json` to `Chrome > 80`. But to my surprise, things like `es6.promise` still appear in the final production bundle.

![image](https://user-images.githubusercontent.com/25521218/84661253-a8a54480-af4c-11ea-9f15-964373901039.png)

After that, I **remove gatsby-plugin-mdx** from my repo and rebuild my project with `Chrome > 80`. This time it works!

![image](https://user-images.githubusercontent.com/25521218/84661882-91b32200-af4d-11ea-8344-052f59bedba5.png)

And I change the `browserslist` back to `defaults`. And here is the result:

![image](https://user-images.githubusercontent.com/25521218/84662228-1ef67680-af4e-11ea-90b0-d049351f1236.png)

### Steps to reproduce

Set your `browserslist` in `package.json` to `Chrome > 80`:

``` json
  "browserslist": [
    "Chrome > 80"
  ],
```

And add `gatsby-plugin-webpack-bundle-analyser-v2` to your plugin list, then build the project.

### Expected result

I guess that Chrome 80+ do not need polyfills for Promise. But they still appear in the production bundle.

I expect to get a "clean" bundle without these redundant polyfills. Just like the screenshot shows.

### Actual result

things like `es6.promise` still appear in the final production bundle.

### Environment

```

  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
  Binaries:
    Node: 12.17.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.13.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.2 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.8.3 - C:\Python38\python.EXE
  Browsers:
    Edge: 44.19041.1.0
  npmPackages:
    gatsby: ^2.21.13 => 2.22.3 
    gatsby-local-autolink-headers: file:plugins/gatsby-local-autolink-headers => 1.0.0 
    gatsby-plugin-advanced-sitemap: ^1.5.2 => 1.5.4 
    gatsby-plugin-catch-links: ^2.3.1 => 2.3.2 
    gatsby-plugin-manifest: ^2.4.2 => 2.4.8 
    gatsby-plugin-material-ui: ^2.1.8 => 2.1.8 
    gatsby-plugin-mdx: ^1.2.6 => 1.2.11 
    gatsby-plugin-offline: ^3.2.1 => 3.2.6 
    gatsby-plugin-react-helmet: ^3.3.1 => 3.3.2 
    gatsby-plugin-webpack-bundle-analyser-v2: ^1.1.10 => 1.1.10 
    gatsby-remark-copy-linked-files: ^2.3.2 => 2.3.3 
    gatsby-remark-table-of-contents: 0.0.9 => 0.0.9 
    gatsby-remark-vscode: ^2.1.1 => 2.1.1 
    gatsby-source-filesystem: ^2.3.1 => 2.3.7 
    gatsby-source-local-git: ^1.0.0 => 1.0.0 
    gatsby-transformer-json: ^2.4.3 => 2.4.3 

```

**Comment by kenny-f** [comment]
I'm seeing the same thing on a new project after running `gatsby new`

**Comment by LekoArts** [comment]
Hi!

I'm closing this as a stale issue as in the meantime Gatsby 4 and related packages were released. You can check our [Framework Version Support Page](https://www.gatsbyjs.com/docs/reference/release-notes/gatsby-version-support/) to see which versions currently receive active support.

Please try the mentioned issue on the latest version (using the `next` tag) and if you still see this problem, open a new bug report. It **must** include a [minimal reproduction](https://gatsby.dev/reproduction).

Thanks!
